iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -38,25 +39,40 @@ class ServiceRegistry
     };
 
     using ReferenceCounter_t = uint64_t;
+
     struct ServiceDescriptionEntry
     {
+        ServiceDescriptionEntry(const capro::ServiceDescription& sd, ReferenceCounter_t count)
+            : serviceDescription(sd)
+            , count(count)
+        {
+        }
+
         capro::ServiceDescription serviceDescription{};
-        ReferenceCounter_t referenceCounter = 0U;
+        ReferenceCounter_t count{0U};
     };
 
     /// @todo #415 should be connected with iox::MAX_NUMBER_OF_SERVICES
+    ///            increase limit to at least >= 1000
     static constexpr uint32_t MAX_SERVICE_DESCRIPTIONS = 100U;
```

Comment:
Why not set it here directly to `iox::MAX_NUMBER_OF_SERVICES` otherwise I may fear that you brake master.

Reply:
Will see whether it works.

Reply:
There are a lot of problems related to this, hence a new issue #1074.